core/lib/mini_merkle_tree/src/lib.rs

```diff
@@ -19,16 +19,26 @@ use zksync_crypto::hasher::{keccak::KeccakHasher, Hasher};
 /// we unlikely to ever hit.
 const MAX_TREE_DEPTH: usize = 32;
 
-/// In-memory Merkle tree of bounded depth (no more than 10).
+/// In-memory Merkle tree of bounded depth (no more than 32).
```

Comment:
Bikeshedding: Technically, it's now not a full Merkle tree, but rather a partial tree, potentially with the tree start trimmed. I'd rename the type `PartialMerkleTree` or `TrimmedMerkleTree` correspondingly.

Reply:
I'm not sure this will help the understanding of what it does.

- It can still be used as a full Merkle tree
- I believe `Mini` is still a good enough descriptor since it highlights a property with a more immediate consequence
- To understand what is meant by `Partial` it will anyway be necessary to read the docs